samples: openthread: add low power mode to CLI (#9885)
nrfconnect/sdk-nrf: samples/openthread/cli/README.rst

* :file:`overlay-ci.conf` - Disables boot banner and shell prompt.
* :file:`overlay-multiprotocol.conf` - Enables Bluetooth LE support in this sample.
* :file:`overlay-tcp.conf` - Enables experimental TCP support in this sample.

> * :file:`overlay-low_power.conf` - Enables low power consumption mode in this sample. Additionally, you need to set :makevar:`DTC_OVERLAY_FILE` to :file:`low_power.overlay`.

Reviewer: ```suggestion
* :file:`overlay-low_power.conf` - Enables low power consumption mode in this sample.
  Additionally, you need to set :makevar:`DTC_OVERLAY_FILE` to :file:`low_power.overlay`.
```

            addresses: [fdde:ad00:beef:0:e0fc:dc28:1d12:8c2]
         Done

Power consumption measurements
==============================

> The Thread CLI sample can be used to perform power consumption measurements for Sleepy End Devices.

Reviewer: ```suggestion
You can use the Thread CLI sample to perform power consumption measurements for Sleepy End Devices.
```
I also noticed we are using the SED acronym without defining it in this sample. Could you change Line 360 "SED" to "Sleepy End Device (SED)", I believe that's the first spot where the term is used. That should cover it.

> After building and flashing with :file:`overlay-low_power.conf` and :file:`low_power.overlay` the device will start regular operation with UART console enabled.

Reviewer: ```suggestion
After building and flashing with :file:`overlay-low_power.conf` and :file:`low_power.overlay`, the device will start regular operation with the UART console enabled.
```
Also add an empty line before this line. Better to separate the instructional part from the descriptive first sentence.

> This allows for easy configuration of the device, specifically the SED polling period or the SSED CSL period, among other relevant parameters.

Reviewer: ```suggestion
This allows for easy configuration of the device, specifically the Sleepy End Device polling period or the Synchronized Sleepy End Device (SSED) CSL period and other relevant parameters.
```

> Then, once the device becomes attached to a Thread Router it will automatically suspend UART operation and power down unused RAM.

Reviewer: ```suggestion
When the device becomes attached to a Thread Router it will automatically suspend UART operation and power down unused RAM.
```
Add an empty line above this line as well.

> From this moment it is not possible to use the CLI to control the device, but it will periodically wake up from deep sleep mode and turn on the radio to receive any messages from its parent.

Reviewer: ```suggestion
In this mode, you can not use the CLI to control the device.
Instead, the device will periodically wake up from deep sleep mode and turn on the radio to receive any messages from its parent.
```

> If the device is connected to a Power Profiler Kit II, detailed power comsumption measurmens can be performed.

Reviewer: ```suggestion
If the device is connected to a Power Profiler Kit II, you can perform detailed power consumption measurements.
```

Author: Do you think we can add some appropriate PPK2 link here?

Reviewer: Absolutely. You can use the \`Power Profiler Kit II (PPK2)\`_ syntax for that.

            addresses: [fdde:ad00:beef:0:e0fc:dc28:1d12:8c2]
         Done

Power consumption measurements
==============================

You can use the Thread CLI sample to perform power consumption measurements for Sleepy End Devices.

After building and flashing with :file:`overlay-low_power.conf` and :file:`low_power.overlay`, the device will start regular operation with the UART console enabled.
This allows for easy configuration of the device, specifically the Sleepy End Device polling period or the Synchronized Sleepy End Device (SSED) CSL period and other relevant parameters.

> When the device becomes attached to a Thread Router it will automatically suspend UART operation and power down unused RAM.

Reviewer: ```suggestion

When the device becomes attached to a Thread Router it will automatically suspend UART operation and power down unused RAM.
```
Another empty line here.
